packages/server/src/fhir/operations/patienteverything.ts

```diff
@@ -74,11 +90,82 @@ export async function getPatientEverything(
     });
   }
 
-  return repo.search({
+  // Get initial bundle of compartment resources
+  const bundle = await repo.search({
     resourceType: 'Patient',
     types,
     filters,
     count: params?._count ?? defaultMaxResults,
     offset: params?._offset,
   });
+
+  // Recursively resolve references
+  await addResolvedReferences(repo, bundle.entry);
+  return bundle;
+}
+
+/**
+ * Recursively resolves references in the given resources.
+ * @param repo - The repository.
+ * @param entries - The initial resources to process.
+ */
+async function addResolvedReferences(repo: Repository, entries: BundleEntry[] | undefined): Promise<void> {
+  const processedRefs = new Set<string>();
+  let page = entries;
+  while (page?.length) {
+    const references = processReferencesFromResources(page, processedRefs);
+    const resolved = await repo.readReferences(references);
+    page = flatMapFilter(resolved, (resource) =>
+      isResource(resource) ? { resource, search: { mode: 'include' } } : undefined
+    );
+    entries?.push(...page);
```

Comment:
Any reason to make a shallow copy here? I think since all these functions are exported and only used locally we know that the page is read only

Reply:
Page is iterated over each loop, but needs to be copied into `entries` as well; what's the alternative?

Reply:
Oh actually I misread what this block is doing, sorry. I thought we were making a copy of `page` in order to avoid potential mutation in `processReferencesFromResources` but we're just copying the result in `entries` before the next iteration